src/problems/peg_game.clj

```diff
@@ -0,0 +1,144 @@
+;;             5 [[0 2] [3 4] [12 8] [14 9]]
+;;             6 [[1 3] [8 7]]
+;;             7 [[2 4] [9 8]]
+;;             8 [[1 4] [6 7]]
+;;             9 [[2 5] [7 8]]
+;;             10 [[3 6] [12 11]]
+;;             11 [[4 7] [13 12]]
+;;             12 [[3 7] [5 8] [10 11] [14 13]]
+;;             13 [[4 8] [11 12]]
+;;             14 [[5 9] [12 13]]))
+
+
+;;Diagonal increase left to right
+;;[(+ (+ vector-index row-number) (+ row-number 1)) (+ vector-index row-number)]
+
+;;Diagonal increase right to left
+;;[(+ (+ vector-index (+ row-number 1)) (+ row-number 2)) (+ vector-index (+ row-number 1))]
+
+;;Diagonal decrease left to right
+;;[(- (- vector-index row-number) (- row-number 1)) (- vector-index row-number)]
+
+;;Diagonal decrease right to left
+;;[(- (- vector-index (- row-number 1)) (- row-number 2)) (- vector-index (- row-number 1))]
+
+(defn num-rows [board]
+  (loop [index 1
+         total-index (count board)
+         current-index 0
+         number-of-rows 0]
+    (if (not= total-index current-index)
+      (recur (+ index 1) total-index (+ current-index index) (+ number-of-rows 1))
+      number-of-rows)))
+
+
+(defn generate-moves [board vector-index row-index row-number number-of-rows hashmap]
+  [
+    (if (> row-index 2) [(- vector-index 2) (- vector-index 1)])
+    (if (< row-index (- row-number 1)) [(+ vector-index 2) (+ vector-index 1)])
+    (if (> row-index 2) [(- (- vector-index row-number) (- row-number 1)) (- vector-index row-number)])
+    (if (< row-index (- row-number 1)) [(- (- vector-index (- row-number 1)) (- row-number 2)) (- vector-index (- row-number 1))])
+    (if (< row-number (- number-of-rows 1)) [(+ (+ vector-index (+ row-number 1)) (+ row-number 2)) (+ vector-index (+ row-number 1))])
+    (if (< row-number (- number-of-rows 1)) [(+ (+ vector-index row-number) (+ row-number 1)) (+ vector-index row-number)])
+  ])
+
+
+
+
+(defn make-moves
+  "makes the moves that are possible for the board given to it"
+  [board]
+  (loop [board board
+         vector-index 0
+         row-index 1
+         row-number 1
+         number-of-rows (num-rows board)
+         hashmap {}]
+    (if (not= vector-index (count board))
+      (if (not= row-index row-number)
+        (recur board (+ vector-index 1) (+ row-index 1) row-number number-of-rows (assoc hashmap vector-index (generate-moves board vector-index row-index row-number number-of-rows hashmap)))
+        (recur board (+ vector-index 1) 1 (+ row-number 1) number-of-rows (assoc hashmap vector-index (generate-moves board vector-index row-index row-number number-of-rows hashmap))))
+      hashmap)))
+
+
+
+;;Need to put the same size or larger peg game as the one you are actually performing the search on.
+
+(def moves
+ (make-moves [1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1]))
+
+
+;; Heuristic perfers empty spaces on the bottom of the board.
+
+(defn heuristic
+  "Heuristic perfers empty spaces on the bottom of the board."
+  [parent child]
```

Comment:
You pass `is-true` as the first argument to this function, which (based on how it's used in search algorithms) is a predicate. Here you call it `parent`, though, which implies it's a state. Luckily you never use it here, because I have a feeling bad things would happen.